poandy/controller/trade.py

```diff
@@ -0,0 +1,45 @@
+from poandy.util.request import RequestSender, RequestType
+from poandy.controller.base import Controller
+
+
+class TradeController(Controller):
+    @classmethod
+    def get_trades(cls, account_id, state="OPEN", instrument=None, count=50):
+        if state not in ["OPEN", "CLOSED", "CLOSE_WHEN_TRADEABLE", "ALL"]:
+            raise ValueError(
+                "state must be a valid TradeStateFilter (OPEN, CLOSED, CLOSE_WHEN_TRADEABLE, or ALL)"
+            )
+        if count < 1 or count > 500:
+            raise ValueError("count must be positive and not exceed 500")
+
+        url = f"{cls._config['base_url']}/v3/accounts/{account_id}/trades"
+        params = {"state": state, "count": count}
+        if instrument:
+            params["instrument"] = instrument
+
+        response = RequestSender.send(url, cls._headers, RequestType.GET, params=params)
+        return (
+            response.json()
+            if response.status_code == 200
+            else response.raise_for_status()
+        )
+
+    @classmethod
+    def get_open_trades(cls, account_id):
+        url = f"{cls._config['base_url']}/v3/accounts/{account_id}/openTrades"
+        response = RequestSender.send(url, cls._headers, RequestType.GET)
+        return (
+            response.json()
+            if response.status_code == 200
+            else response.raise_for_status()
+        )
+
+    @classmethod
+    def get_open_trades(cls, account_id):
```

Comment:
This method is the same as the one above. Please remove.

Reply:
Please also implement the method to GET the specific details of a trade and its tests

Reply:
the duplicate `get_open_trades()` method is removed in commit https://github.com/QAM-ATC/poandy/pull/18/commits/815aabe4e9220e1fb133d6799df7f9ea310e5a16

Are you referring to the endpoint `https://api-fxtrade.oanda.com/v3/accounts/<ACCOUNT>/trades/<tradeSpecifier>`? I checked and it seems that the specific details of a trade also appear in the other endpoints, so I feel a bit redundant to implement GET request for this endpoint. But sure I can add the method for that endpoint.

Reply:
a more thorough test is to place a market order wait around 5 seconds and obtain a trade confirmation. however, this also means that the test can only pass during trading hours (weekdays 24hours). if you have capacity :)